src/utils.rs

```diff
@@ -162,37 +162,37 @@ pub(crate) fn random_plusminus_by_size_with_minimum<R: RngCore + CryptoRng>(
 /// [`Transcript`].
 pub(crate) fn plusminus_bn_random_from_transcript(
     transcript: &mut Transcript,
-    n: &BigNumber,
+    _n: &BigNumber,
 ) -> BigNumber {
     let mut is_neg_byte = vec![0u8; 1];
     transcript.challenge_bytes(b"sampling negation bit", is_neg_byte.as_mut_slice());
     let is_neg: bool = is_neg_byte[0] & 1 == 1;
 
     // The sampling method samples from the open interval, so add 1 to sample from
     // the _closed_ interval we want here.
-    let open_interval_max = n + 1;
+    let q = k256_order();
+    let open_interval_max = &q + 1;
     let b = positive_bn_random_from_transcript(transcript, &open_interval_max);
     match is_neg {
         true => -b,
         false => b,
     }
 }
 
-/// Derive a deterministic pseduorandom value in `[0, n)` from the
-/// [`Transcript`].
 pub(crate) fn positive_bn_random_from_transcript(
     transcript: &mut Transcript,
-    n: &BigNumber,
+    _n: &BigNumber,
```

Comment:
Sadly, since pimod and pisch are weirdos, I think we do still need to take this as a parameter.